### DISCUSSION TITLE
GEOG_MATCH Interpolation Clarification

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#1293] [Configuration] [answered]
Hi wonderful MET team,

I am interested in using the GEOG_MATCH interpolation method, but I am a bit confused as to how it works. GEOG_MATCH is defined in the UG as "GEOG_MATCH for the nearest grid point where the land/sea mask and geography criteria are satisfied." What exactly is meant by geography criteria? Where is it defined?

Thank you!
Lindsay

**Comment by j-opatz** [comment, marked as answer]
Hi Lindsay,

And thank you for your question. The geography criteria you mentioned is the masking that's being applied by the users to a given field. More specifically, as the User's Guide states, it's the "land/sea mask", accessible via the `land_mask` dictionary, and the "geography criteria", which is more broadly the `mask` dictionary.

On review of the GEOG_MATCH entry I tend to agree with your assessment; it's worded a bit broadly and could stand to be more fully defined.

I'm going to tag @JohnHalleyGotway here for three points: (1) to correct anything I may have stated incorrectly above, (2) to add any details I may have missed (does the `topo_mask` dictionary also apply to this?) and possibly providing a snippet of the code where this logic is performed, and (3) getting his assessment on a more robust definition of the GEOG_MATCH entry, which I can take care of with the info provided in this question.

-John O.

**Reply by JohnHalleyGotway** [reply]
@lindsayrblank I see you have a question about the GEOG_MATCH "interpolation" method. In the [User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html), its described as:
```
The “method” entry specifies the interpolation procedure to be applied to the points in the box:
...
- GEOG_MATCH for the nearest grid point where the land/sea mask and geography criteria are satisfied.
```
In the default Point-Stat [config file](https://github.com/dtcenter/MET/blob/main_v10.0/met/data/config/PointStatConfig_default), the relevant dictionaries are [message_type_group_map](https://github.com/dtcenter/MET/blob/5e1f551d86fcb5b731e3a83c0b043dbd7ef3c7aa/met/data/config/PointStatConfig_default#L107) and [land_mask](https://github.com/dtcenter/MET/blob/5e1f551d86fcb5b731e3a83c0b043dbd7ef3c7aa/met/data/config/PointStatConfig_default#L160).

The "LANDSF" and "WATERSF" entries in message_type_group_map define the message types that should be interpreted as being at the surface over land and water, respectively. Multiple entries are provided in a comma-separated list.

The "land_mask" dictionary specifies how to read a land/sea mask of gridded data (see this [description](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html?highlight=land_mask#land-mask)).

These settings are required to support the use of the GEOG_MATCH interpolation method. For each point observation processed, its message type determines how it's processed. For "land" obs, the nearby model water grid points are ignored. For "water" obs, the nearby model land grid points are ignored. So this is how we can ensure that land obs are compared to land model grid points and water obs are compared to water model grid points.

This [interp_geog_match()](https://github.com/dtcenter/MET/blob/5e1f551d86fcb5b731e3a83c0b043dbd7ef3c7aa/met/src/basic/vx_util/interp_util.cc#L638) function finds the nearest model grid point within the interpolation area whose land/water type (defined by "land_mask") matches that of the observation (defined by "message_type_group_map").

Hope that helps clarify.

**Reply by lindsayrblank** [reply]
Hi John and John, 

Thank you for the clarification. That really helps! 

Best,
Lindsay

**Reply by JohnHalleyGotway** [reply]
Thanks for confirming. I'll go ahead and mark this question as "answered" and "lock" the conversation to prevent future posts. That's our way of encouraging users to create new discussions for new questions.

Thanks,
John HG
